libnetwork/iptables/iptables.go

```diff
@@ -424,3 +424,57 @@ func (iptable IPTable) EnsureJumpRule(fromChain, toChain string, rule ...string)
 	}
 	return nil
 }
+
+type Rule struct {
+	IPVer IPVersion
+	Table Table
+	Chain string
+	Args  []string
+}
+
+// Exists returns true if the rule exists in the kernel.
+func (r Rule) Exists() bool {
```

Comment:
Note; I went looking where `Exists()` was used outside of this type itself (i.e., as part of functions like `Append()`), and it looks like it's only used in tests to assert that some actions happened.


If we're gonna work on follow-ups, we could look if we need `Exists()` to be part of this interface, or if it would be a dedicated test-utility. We could still split it between a non-exported one (for the internal use) and an exported one, to make it easier to discover where the exported one is used.